### PR TITLE
fix: match errors to respective http status code

### PR DESCRIPTION
**Pull request by blaketingyu**

### CODE REVIEW

File: backend/src/user/user.controller.ts [hunk 1]
```diff
@@ -66,8 +70,11 @@ export class UserController {
   @ApiOperation({ summary: "Edit data of specified user" })
   @ApiOkResponse({ description: "The resource was updated successfully" })
   @ApiNotFoundResponse({ description: "Resource not found" })
-  @ApiForbiddenResponse({ description: "Unauthorized Request" })
+  @ApiUnauthorizedResponse({ description: "Unauthorized Request" })
+  @ApiForbiddenResponse({ description: "Unauthorized Request, client does not have access rights to the requested content" })
```
Comment (aidanaden) [inline backend/src/user/user.controller.ts:74]: what's the difference between `ApiUnauthorizedResponse` and `ApiForbiddenResponse`?

**Review by aidanaden — APPROVED** [review]
Mostly LGTM! just 1 question

File: backend/src/user/user.controller.ts (hunk 1 above)
Comment (blaketingyu) [reply backend/src/user/user.controller.ts:74]: `ApiUnauthorizedResponse` is when the client provides no credentials or invalid credentials. `ApiForbiddenResponse` is the status code to return when a client has valid credentials but not enough privileges to perform an action on a resource. I guess maybe I could elaborate more in the `description`.

**Review by yusufaine — APPROVED** [review]
LGTM, good to clarify what you explained to Aidan in the error description itself too!

File: backend/src/user/user.controller.ts (hunk 1 above)
Comment (aidanaden) [reply backend/src/user/user.controller.ts]: ah okay can just add it in the swagger description!

**Pull request merged**